security/pkg/nodeagent/sds/sdsservice.go

```diff
@@ -267,6 +287,38 @@ func (s *sdsservice) FetchSecrets(ctx context.Context, discReq *xdsapi.Discovery
 	return sdsDiscoveryResponse(secret, discReq.Node.Id)
 }
 
+func (s *sdsservice) Stop() {
+	s.closing <- true
+}
+
+func (s *sdsservice) clearStaledClientsJob() {
+	s.ticker = time.NewTicker(s.tickerInterval)
+	for {
+		select {
+		case <-s.ticker.C:
+			s.clearStaledClients()
+		case <-s.closing:
+			if s.ticker != nil {
+				s.ticker.Stop()
+			}
+		}
+	}
+}
+
+func (s *sdsservice) clearStaledClients() {
+	log.Debug("start staled connection cleanup job")
+	sdsClientsMutex.Lock()
+	defer sdsClientsMutex.Unlock()
+
+	len := len(staledClientKeys)
+	for i := 0; i < len; i++ {
```

Comment:
nit: confusing to assign to `len` which is a function already. Can probably just use `range`

Reply:
renamed(here take a snapshot of staledClientKeys by taking its current length).